domain/service/json.go

```diff
@@ -15,10 +15,38 @@ package service
 
 import (
 	"encoding/json"
+
+	"github.com/control-center/serviced/domain"
 	"github.com/control-center/serviced/domain/servicedefinition"
 	"github.com/zenoss/glog"
 )
 
+type service Service
+
+func (s *Service) UnmarshalJSON(b []byte) error {
+	svc := service{}
+	if err := json.Unmarshal(b, &svc); err == nil {
+		*s = Service(svc)
+	} else {
+		return err
+	}
+	if len(s.Commands) > 0 {
+		s.Runs = nil
+		return nil
+	}
+	if len(s.Runs) > 0 {
+		s.Commands = make(map[string]domain.Command)
+		for k, v := range s.Runs {
+			s.Commands[k] = domain.Command{
+				Command:         v,
+				CommitOnSuccess: false,
```

Comment:
For backwards compatibility, shouldn't the value be `true`?

The use-case I'm thinking about is as follows - customer has 1.0.x/5.0.x installed, they upgrade to CC 1.1 but don't upgrade RM 5.0.x. The 5.0.x services should work like they did before (which was CommitOnSuccess=true, right?)

For RM 5.1 service definitions, we don't really need to worry about the value in this method because those services will explicitly define the right values on a case-by-case basis. 
Ditto for upgrades from RM 5.0.x to RM 5.1.